players/g6_player/__init__.py

```diff
@@ -25,48 +28,162 @@ def __init__(
         self.logger = logger
         self.maximum_door_frequency = maximum_door_frequency
         self.radius = radius
-        self.known_target = False
-        # Data structure to hold state information about the doors inside the radius
-        self.curr_maze = {}
+        self.maze_boundary_x = 201
+        self.maze_boundary_y = 201
+        # player_maze[x][y][cycle][door_direction] = boolean door open status
+        self.player_maze_freq = np.zeros((self.maze_boundary_x, self.maze_boundary_y, self.maximum_door_frequency, 4))
+        # player_maze_path_intermediate[x][y][4] = frequency each door is open
+        self.player_maze_path_intermediate = np.zeros((self.maze_boundary_x, self.maze_boundary_y, 4))
+        # player_maze_path[x][y][4] = frequency each path is open
+        self.player_maze_path = np.zeros((self.maze_boundary_x, self.maze_boundary_y, 4))
         self.turn = 0
+        self.cycle = 0
+        self.move_history = [np.array([0,0])]
 
     def move(self, current_percept: TimingMazeState) -> int:
         # Bug: Move Enum is not an int and will not be accepted as a move
         self.turn += 1
+        self.cycle = self.turn % self.maximum_door_frequency
+        if self.cycle == 0:
+            self.cycle = self.maximum_door_frequency
+        # Adjust drone coordinates to be positive when going right and center on the 200,200 grid.
+        adjusted_x = (current_percept.start_x * -1) + 100
```

Comment:
I think the coordinates increase when going right - all the way to the left is 0 and all the way to the right is 99. 
For example, if we start at position `X` but have moved left 5 times, our current position is `X-5`, meaning that from the agent's perspective, `start_x` should be positive 5.
Am I misunderstanding?